guvnor-ala/guvnor-ala-services-api/src/main/java/org/guvnor/ala/services/api/backend/RuntimeProvisioningServiceBackend.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright 2016 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.guvnor.ala.services.api.backend;
+
+import java.util.List;
+import org.guvnor.ala.config.ProviderConfig;
+
+import org.jboss.errai.bus.server.annotations.Remote;
+import org.guvnor.ala.config.RuntimeConfig;
+import org.guvnor.ala.runtime.Runtime;
+import org.guvnor.ala.runtime.providers.Provider;
+import org.guvnor.ala.runtime.providers.ProviderType;
+import org.guvnor.ala.services.exceptions.BusinessException;
+
+/*
+ * Runtime Provisioning Service Backend interface. It allows us to register and interact with different Provisioning Providers and
+ *  to create new Runtimes on these providers.
+ * Backend @Remote implementation to be used in CDI environments with Errai
+ */
+@Remote
+public interface RuntimeProvisioningServiceBackend {
+
+    /*
+     * Get all the registered ProviderTypes
+     * @return ProviderTypeList containing all the registered provider types
+     * @see ProviderTypeList
+     * @throw BusinessException in case of an internal exception
+     */
+    List<ProviderType> getAllProviderTypes() throws BusinessException;
```

Comment:
won't it need any paging support? Not sure what amount of providers we might have so maybe it's not as important.
And maybe rename the method to getProviderTypes (skipping the All)